frontend/app/routes/protected/person-case/current-name.tsx

```diff
@@ -162,12 +189,32 @@ export async function action({ context, request }: Route.ActionArgs) {
 
 export default function CurrentName({ loaderData, actionData, params }: Route.ComponentProps) {
   const { t } = useTranslation(handle.i18nNamespace);
+  const [sameName, setSameName] = useState<boolean | undefined>(loaderData.defaultFormValues.preferredSameAsDocumentName);
   const [requireDoc, setRequireDoc] = useState<boolean | undefined>(loaderData.defaultFormValues.supportingDocumentsRequired);
   const fetcherKey = useId();
   const fetcher = useFetcher<Info['actionData']>({ key: fetcherKey });
   const isSubmitting = fetcher.state !== 'idle';
   const errors = fetcher.data?.errors;
 
+  function handleSameNameChanged(event: ChangeEvent<HTMLInputElement>) {
+    setSameName(event.target.value === REQUIRE_OPTIONS.yes);
+  }
+
+  const nameOptions: InputRadiosProps['options'] = [
+    {
+      children: t('gcweb:input-option.yes'),
+      value: REQUIRE_OPTIONS.yes,
+      defaultChecked: sameName === true,
```

Comment:
If you allow `nameName` to be a simple boolean (ie: not undefined) then this can just be: `defaultChecked: sameName`